activerecord/test/cases/strict_loading_test.rb

```diff
@@ -256,6 +257,24 @@ def test_strict_loading_with_has_many
     end
   end
 
+  def test_strict_loading_with_has_many_and_query_method
+    with_strict_loading_by_default(Developer) do
+      dev = Developer.preload(:audit_logs).first
+
+      assert_raises ActiveRecord::StrictLoadingViolationError do
+        dev.audit_logs.limit(1).first
+      end
+    end
+  end
+
+  def test_strict_loading_with_has_many_and_query_method_in_scope
```

Comment:
Added this as a regression to ensure that preloaded associations with a query method don't raise.